### PR TITLE
improvement: added combatibility check between munlib and runtime

### LINKED ISSUE #188: Implement compatibility checks between munlib and runtime
**Problem**

Currently, the runtime tries to load a `*.munlib` with the assumption that the ABI memory layout of the `*.munlib` matches its own ABI version. This can cause undefined behaviour when it tries to load a `*.munlib` with a different ABI version.

**Solution**

By including an ABI version in the `*.munlib`, the runtime can gracefully fail - reporting an error to the end user.

In the future - once we support backwards compatibility, we can even support loading of several different ABI versions using this functionality.

**Tasks**

- [ ] add an integer `ABI_VERSION` constant to the `mun_abi` crate.
- [ ] [generate](https://github.com/mun-lang/mun/blob/0d6e36d64376289a8969c62cefdc080ee99c2a51/crates/mun_codegen/src/code_gen/symbols.rs#L291) a `get_version` IR function
- [ ] [export](https://github.com/mun-lang/mun/blob/c17448d77f9751e74ce90ea49cd8503b35459d8e/crates/mun_codegen/src/code_gen/linker.rs#L176) the `get_version` function
- [ ] in the [runtime](https://github.com/mun-lang/mun/blob/master/crates/mun_runtime/src/assembly.rs#L35), call the `get_version` function and verify that the runtime's `mun_abi::ABI_VERSION` is compatible.
- [ ] (bonus) add a constant with the `get_version` function name (and other API functions) to the `mun_abi` crate and use that constant in both the `mun_codegen` and `mun_runtime` crates - to avoid discrepancies.

**Good first issue process:**

- [ ] [read our contributing guidelines](https://github.com/mun-lang/mun/blob/master/CONTRIBUTING.md)
- [ ] claim this issue (assign yourself or comment below)
- [ ] implement the feature and start a Pull Request with in the description: closes #188. 

If this is your first PR, welcome 🎉 😄

### PR DESCRIPTION
**Pull request by MineBill**
This PR closes #188

### CODE REVIEW

File: crates/mun_abi/src/lib.rs [hunk 1]
```diff
@@ -25,6 +25,15 @@ pub mod prelude {
     };
 }
 
+/// Defines the current abi version
+pub const ABI_VERSION: i32 = 00_20_00;
```
Comment (Wodann) [inline crates/mun_abi/src/lib.rs:29]: Did you have a reason in mind for choosing a signed integer (`i32`) over an unsigned integer (`u32`)?

**Comment by Wodann** [comment]
Nice work! 👏 

I have one [question](https://github.com/mun-lang/mun/pull/226/files#r440114416)

File: crates/mun_abi/src/lib.rs (hunk 1 above)
Comment (MineBill) [reply crates/mun_abi/src/lib.rs:29]: Not really, should i change it to a u32?

Comment (Wodann) [reply crates/mun_abi/src/lib.rs:29]: I don't think we'll ever hit the maximum number of versions possible with `i32`, but still I think that `u32` would be more correct - as I don't think we'll ever use negative version numbers.

Comment (Wodann) [inline crates/mun_abi/src/lib.rs:29]:
```suggestion
#[allow(clippy::zero-prefixed-literal)]
pub const ABI_VERSION: i32 = 00_02_00;
```

Comment (Wodann) [reply crates/mun_abi/src/lib.rs:29]: I am not sure whether @baszalmstra agrees, but I think we can ignore this clippy warning. I like the general formatting of the version number into major, minor, patch.

I did change minor `20` to `02`

**Pull request merged**